repos/system_upgrade/common/actors/multipath/config_reader/tests/test_multipath_conf_read_8to9.py

```diff
@@ -100,45 +104,36 @@ def test_parse_config():
         assert_config(config, expected_data)
 
 
-def test_get_facts_missing_dir(monkeypatch):
-    monkeypatch.setattr(multipathconfread, '_parse_config_orig', multipathconfread._parse_config, raising=False)
-    monkeypatch.setattr(multipathconfread, '_parse_config', mock_parse_config)
-
-    facts = multipathconfread.get_multipath_conf_facts(os.path.join(TEST_DIR, 'missing_dir.conf'))
-    assert facts
-    assert len(facts.configs) == 1
-    assert_config(facts.configs[0], missing_dir_conf)
-
-
-def test_get_facts_empty_dir(monkeypatch):
+@pytest.mark.parametrize(
+    ('primary_config', 'expected_configs'),
+    [
+        ('missing_dir.conf', [missing_dir_conf]),
+        ('empty_dir.conf', [empty_dir_conf]),
+        ('not_set_dir.conf', [not_set_dir_conf, empty1_conf, nothing_important_conf]),
+        ('set_in_dir.conf', [set_in_dir_conf, all_true_conf]),
+    ]
+)
+def test_get_facts_missing_dir(monkeypatch, primary_config, expected_configs):
     monkeypatch.setattr(multipathconfread, '_parse_config_orig', multipathconfread._parse_config, raising=False)
     monkeypatch.setattr(multipathconfread, '_parse_config', mock_parse_config)
+    monkeypatch.setattr(multipathconfread, 'is_processable', lambda: True)
 
-    facts = multipathconfread.get_multipath_conf_facts(os.path.join(TEST_DIR, 'empty_dir.conf'))
-    assert facts
-    assert len(facts.configs) == 1
-    assert_config(facts.configs[0], empty_dir_conf)
+    produce_mock = produce_mocked()
+    monkeypatch.setattr(api, 'produce', produce_mock)
 
+    actor_mock = CurrentActorMocked()
```

Comment:
Let's rather set the IPU path, so if there are any changes in future about defaults, the test is not negatively impacted.

```suggestion
    actor_mock = CurrentActorMocked(src_ver='8.10', dst_ver='9.6')
```

Note that there is not a test for 9 -> 10 checking that 8to9 msg is not produced. MultipathInfo should be produced in both cases.

Reply:
I have:
- made the tests explicit for the 8>9 upgrade path
- added a new test for the 9>10 upgrade patch that checks whether only `MultipathInfo` is produced